samples/msal-node-samples/refresh-token/adalCustomCache.js

```diff
@@ -0,0 +1,114 @@
+var _ = require('underscore');
+var fs = require('fs');
+
+/**
+ * Constructs a new token cache on disk.
+ * This effectively implements the ADAL TokenCache interface.
+ * @constructor
+ */
+function DiskCache(cacheLocation) {
+    this._cacheLocation = cacheLocation;
+    this._entries = [];
+}
+
+/**
+ * Removes a collection of entries from the cache in a single batch operation.
+ * @param {Array} entries An array of cache entries to remove.
+ * @param {Function} callback This function is called when the operation is complete. Any error is provided as the first parameter.
+ */
+DiskCache.prototype.remove = function (entries, callback) {
+    const cachedEntries = _readCache.call(this);
+
+    var updatedEntries = _.filter(cachedEntries, function (element) {
+        if (_.findWhere(entries, element)) {
+            return false;
+        }
+        return true;
+    });
+
+    _writeCache.call(this, updatedEntries);
+    callback();
+};
+
+/**
+ * Adds a collection of entries to the cache in a single batch operation.
+ * @param {Array} entries  An array of entries to add to the cache.
+ * @param {Function} callback This function is called when the operation is complete. Any error is provided as the first parameter.
+ */
+DiskCache.prototype.add = function (entries, callback) {
+    const cachedEntries = _readCache.call(this);
+
+    // Remove any entries that are duplicates of the existing
+    // cache elements.
+    _.each(cachedEntries, function (element) {
+        _.each(entries, function (addElement, index) {
+            if (entriesHaveEqualHashKeys(element, addElement)) {
+                entries[index] = null;
+            }
+        });
+    });
+
+    // Add the new entries to the end of the cache.
+    entries = _.compact(entries);
```

Comment:
It's a bad idea to reassign function parameters. I think you should you create a new variable here.

https://medium.com/the-andela-way/its-bad-practice-to-reassign-function-params-c37f289a5dc4

var compactedEntries = _.compact(entries);

Reply:
That's good catch -updated!